src/vm/callingconvention.h

```diff
@@ -1701,6 +1701,17 @@ inline BOOL HasRetBuffArg(MetaSig * pSig)
     return argit.HasRetBuffArg();
 }
 
+#ifdef UNIX_X86_ABI
+inline BOOL HasRetBuffArgUnmanagedFixup(MetaSig * pSig)
+{
+    WRAPPER_NO_CONTRACT;
+    CorElementType type = pSig->GetReturnTypeNormalized();
+
+    // For UNIX_X86_ABI and unmanaged function, we always need RetBuf if the return type is VALUETYPE
+    return type == ELEMENT_TYPE_VALUETYPE;
```

Comment:
Yes, but then it does a ton of stuff in the `ELEMENT_TYPE_VALUETYPE` case.

Enums are encoded as ELEMENT_TYPE_VALUETYPE in signatures, but they are treated as primitive types for calling convention. `pSig->GetReturnTypeHandleThrowing().GetVerifierCorElementType()` will get the enum treated as primitive type.